### PR TITLE
Extend recommended plugin rules in all configs

### PR DESCRIPTION
**Pull request by rekmarks**
This PR extends the recommended rules of the corresponding plugin for the `mocha`, `nodejs`, and `typescript` configs.

The `nodejs` recommended rules added a single, deprecated core rule, which we now disable in our config.

The `mocha` recommended rules adds [`mocha/no-exports`](https://github.com/lo1tuma/eslint-plugin-mocha/blob/master/docs/rules/no-exports.md), which seems good.

The `typescript` recommended rules adds a bunch of previously absent rules to the snapshot for that config, however almost all of them are already disabled in our base config. In f30711b, I removed some extraneous rules, which you'll notice didn't affect the snapshot.

Following this PR, I will remove all rules that we specify that are identical to those we find in the plugins' recommended rule sets. Then, we can publish a new major version of this package and start using it.

### CODE REVIEW

File: packages/typescript/rules-snapshot.json [hunk 1]
```diff
@@ -110,16 +112,26 @@
...
+  "no-setter-return": "off",
   "no-this-before-super": "off",
   "no-undef": "off",
   "no-unreachable": "off",
+  "no-unsafe-negation": "off",
   "no-unused-expressions": "off",
   "no-unused-vars": "off",
   "no-use-before-define": "off",
   "no-useless-constructor": "off",
+  "no-var": "error",
```
Comment (Gudahtt) [inline packages/typescript/rules-snapshot.json:131]: These four seem new :thinking: These aren't even TypeScript rules. What's going on here?

**Review by Gudahtt — COMMENTED** [review]
LGTM aside from the one question I had

File: packages/typescript/rules-snapshot.json (hunk 1 above)
Comment (rekmarks) [reply packages/typescript/rules-snapshot.json:131]: These come from the `plugin:@typescript-eslint/eslint-recommended` config ([source](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/src/configs/eslint-recommended.ts)), which is extended by `plugin:@typescript-eslint/recommended` (https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/src/configs/recommended.ts)).

They discuss it in their readme here: https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin/src/configs#eslint-recommended

Comment (Gudahtt) [reply packages/typescript/rules-snapshot.json:131]: Got it, thanks. Makes sense.

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Pull request merged**